Guide-for-using-MongoDB-and-deploying-to-Heroku.md

```diff
@@ -0,0 +1,114 @@
+##Guide for using MongoDB and deploying to Heroku
+
+In this guide lets see how to work with MongoDB locally and with mLab for deploying it to Heroku. Alternatively you can also use mLab add-on in Heroku , It is free but it may require your Credit card Details. So, if you are not intrested in providing your credit card details you can go with [mLab](https://mlab.com) website. 
+
+#### Setting up a free account on Heroku and mLab:
+
+Sign up for [Heroku](https://signup.heroku.com/) and [mLab](https://mlab.com/signup/)
+
+
+####Starting your Mongodb(Locally): 
+
+```mongod --port 27017 --dbpath=./data``` _(Create a directory named 'data' if it doesn't exist)_
+
+Now your Db is running at-
+
+`mongodb://localhost:27017/my_database_name`
+
+>If you are using C9 and If you are having trouble starting your DB in C9 refer to this [page](https://community.c9.io/t/setting-up-mongodb/1717)
+
+####Starting your Mongodb(mLab):
+1. After Creating your mLab Account, click on **Create new** button and select Single-node -> Sandbox to get the free Db and give your database a name (i've created a db named 'food' for this as an example) .
+2. Now a database is created with the name 'food',You can create a new collection of your own.
+3. Finally Add a User/Users who can access this Database,While Adding a user it will ask for username and password which are used to access the Database.
+
+Now your DB is running at url something like this -
+
+` mongodb://username:password@ds01316.mlab.com:1316/food `
+
+
+where username and password are the details that you've given when you added a user.
+
+>You can find your DB url at [https://mlab.com/databases/your_database_name](https://mlab.com/databases/your_database_name)
+
+####Making a Connection with MongoDB in Node.js (While DB is running in your Local System):
+
+To work with the database, First you need to create a connection. In this section, we will be using MongoDB’s native Node.js driver to create the connection with the MongoDB server. To install the mongodb native drivers, use the npm command to install the mongodb module. After that, run the following command in your project directory.
+
+``` npm install mongodb```
+
+Basic Code for connecting to MongoDB
+
+```
```

Comment:
See, the multiline code block should have ended way earlier, but it's ending now, because you intend to start one here.

Also, set `js` as syntax.